packages/p2p-v2/src/pages/my-ads/components/PreferredCountriesSelector/__tests__/PreferredCountriesSelector.spec.tsx

```diff
@@ -0,0 +1,116 @@
+                            display_name: 'Alipay ID',
+                            required: 1,
+                            type: 'text',
+                        },
+                        instructions: {
+                            display_name: 'Instructions',
+                            required: 0,
+                            type: 'memo',
+                        },
+                    },
+                    type: 'ewallet',
+                },
+            },
+        },
+        countryB: {
+            country_name: 'countryB',
+            cross_border_ads_enabled: 1,
+            fixed_rate_adverts: 'enabled',
+            float_rate_adverts: 'disabled',
+            float_rate_offset_limit: 10,
+            local_currency: 'CA',
+            payment_methods: {
+                alipay: {
+                    display_name: 'Alipay',
+                    fields: {
+                        account: {
+                            display_name: 'Alipay ID',
+                            required: 1,
+                            type: 'text',
+                        },
+                        instructions: {
+                            display_name: 'Instructions',
+                            required: 0,
+                            type: 'memo',
+                        },
+                    },
+                    type: 'ewallet',
+                },
+            },
+        },
+    },
+    type: 'preferredCountries',
+};
+
+describe('PreferredCountriesSelector', () => {
+    it('should render the component as expected with given props', () => {
+        render(<PreferredCountriesSelector {...mockProps} />);
+        expect(screen.getByText('All countries')).toBeInTheDocument();
+        expect(screen.getByText('Preferred countries')).toBeInTheDocument();
+    });
+    it('should open the modal when the preferred countries field is clicked', () => {
+        render(<PreferredCountriesSelector {...mockProps} />);
+        const element = screen.getByText('All countries');
+        userEvent.click(element);
+        expect(screen.getByRole('button', { name: 'Apply' })).toBeInTheDocument();
+    });
+    it('should close the modal when the apply button is clicked', () => {
+        render(<PreferredCountriesSelector {...mockProps} />);
+        const element = screen.getByText('All countries');
+        userEvent.click(element);
+        const applyButton = screen.getByRole('button', { name: 'Apply' });
+        userEvent.click(applyButton);
+        expect(mockFn).toHaveBeenCalledWith('preferred-countries', ['countryA', 'countryB']);
+        expect(screen.queryByRole('button', { name: 'Apply' })).not.toBeInTheDocument();
+    });
+    it('should close the modal when the close icon is clicked', () => {
+        render(<PreferredCountriesSelector {...mockProps} />);
+        const element = screen.getByText('All countries');
+        userEvent.click(element);
+        const closeButton = screen.getByTestId('dt-close-icon');
```

Comment:
shouldn't the test id be dt_close_icon? @shayan-deriv what convention do you guys follow when adding testids?

Reply:
@ameerul-deriv for now we have both cases but we have a card to only use `dt_` but the thing is even in deriv-app we have both cases.

Reply:
alright thanks man 👍 @nada-deriv you can resolve this